mock_ros2_project/subscriber_node.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+"""
+ROS2 Subscriber Node Example
+
+This node demonstrates how to create a simple subscriber that listens
+to string messages and processes them to show basic pub/sub communication.
+"""
+
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import String
+
+
+class SubscriberNode(Node):
+    """Simple subscriber node that listens to string messages.
+    
+    This node creates a subscription to receive and process string messages
+    from the 'send_topic' topic.
+    """
+    
+    def __init__(self) -> None:
+        """Initialize the subscriber node."""
+        super().__init__('subscriber_node')
+        
+        # Create subscription to receive string messages
+        self.subscription = self.create_subscription(
+            String,
+            'send_topic',  # Fixed: Match publisher topic name
+            self.listener_callback,
+            10
+        )
+        self.subscription  # prevent unused variable warning
```

Comment:
Instead of the extra line to silence the unused-variable warning, you can simply prefix the attribute with an underscore (`self._subscription`) or keep the attribute for later use (e.g., to explicitly destroy it). This removes a no-op line and clarifies intent.

```suggestion
        self._subscription = self.create_subscription(
            String,
            'send_topic',  # Fixed: Match publisher topic name
            self.listener_callback,
            10
        )
```